README.md

```diff
@@ -201,13 +206,69 @@ Sample response :
 
 ```
 
-  
-#### Detokenize
+[Upsert call example](https://github.com/skyflowapi/skyflow-go/blob/main/samples/vault-api/upsert.go):
 
-In order to retrieve data from your vault using tokens that you have previously generated for that data, you can use the **detokenize(records map[string]interface{})** method. The first parameter must have a `records` key that takes an array of tokens to be fetched from the vault, as shown below.
+```go
+package main
 
+import (
+    "fmt"
+    Skyflow "github.com/skyflowapi/skyflow-go/skyflow/client"
+    "github.com/skyflowapi/skyflow-go/skyflow/common"
+)
+
+func main() {
+
+    //Initialize the SkyflowClient.
+
+    var records = make(map[string] interface {})
+    var record = make(map[string] interface {})
+    record["table"] = "cards"
+    var fields = make(map[string] interface {})
+    fields["cardNumber"] = "411111111111"
+    fields["fullname"] = "name"
+    record["fields"] = fields
+    var recordsArray[] interface {}
+    recordsArray = append(recordsArray, record)
+    records["records"] = recordsArray
 
+    //Create an upsert array.
 
+    var upsertArray []common.UpsertOptions
+    var upsertOption = common.UpsertOptions{Table:"cards",Column:"cardNumber"}
+    upsertArray = append(upsertArray,upsertOption)
+
+    var options = common.InsertOptions {
+        Tokens: true
+        Upsert: upsertArray
+    }
+
+    res, err: = skyflowClient.Insert(records, options)
+
+    if err == nil {
+        fmt.Println(res.Records)
+    }
+}
+```
+
+Sample response :
+
+```json
+{
+  "records": [
+    {
+      "table": "cards",
+      "fields": {
+        "cardNumber": "f37186-e7e2-466f-91e5-48e2bcbc1",
+        "fullname": "1989cb56-63a-4482-adf-1f74cd1a5"
+      }
+    }
+  ]
+}
+```
+
+#### Detokenize
+In order to retrieve data from your vault using tokens that you have previously generated for that data, you can use the **Detokenize(records map[string]interface{})** method. The first parameter must have a `records` key that takes an array of tokens to be fetched from the vault, as shown below.
```

Comment:
I'd update 

"In order to retrieve data from your vault using tokens that you have previously generated for that data, you can use the **Detokenize(records map[string]interface{})** method," 

to

To retrieve tokens from your vault,  you can use...

The records parameter takes a map that has an array of SkyflowIDs to return, as shown below: